frontend/src/components/Agreements/AgreementDetails/AgreementHistoryPanel.jsx

```diff
@@ -0,0 +1,173 @@
+import PropTypes from "prop-types";
+import { useGetAgreementHistoryByIdQuery } from "../../../api/opsAPI";
+import { historyData } from "../../../pages/agreements/details/data";
+import LogItem from "../../UI/LogItem";
+import { convertCodeForDisplay } from "../../../helpers/utils";
+
+const findObjectTitle = (historyItem) => {
+    if (historyItem.class_name === "BudgetLineItem") {
+        return historyItem.event_details.line_description;
+    } else {
+        return historyItem.event_details.name;
+    }
+};
+
+const summaryMessage = (historyItem) => {
+    const className = convertCodeForDisplay("className", historyItem.class_name);
+    const user_name = historyItem.created_by_user_full_name;
+    const objectTitle = findObjectTitle(historyItem);
+    const classAndTitle = `${className}, "${objectTitle}",`;
+    if (historyItem.event_type === "NEW") {
+        return `New ${classAndTitle} created by ${user_name}.`;
+    } else if (historyItem.event_type === "UPDATED") {
+        return `${classAndTitle} updated by ${user_name}.`;
+    } else if (historyItem.event_type === "DELETED") {
+        return `${classAndTitle} deleted by ${user_name}.`;
+    }
+    return `${className} ${historyItem.event_type} ${user_name}`;
+};
+
+const getPropertyLabel = (className, fieldName) => {
+    if (className === "BudgetLineItem") return convertCodeForDisplay("budgetLineItemPropertyLabels", fieldName);
+    return convertCodeForDisplay("agreementPropertyLabels", fieldName);
+};
+
+const usersToNames = (users) => {
+    return users.map((user) => user.full_name);
+};
+
+const ChangesDetails = ({ historyItem }) => {
+    const changes = historyItem.changes;
+    const eventType = historyItem.event_type;
+    if (eventType != "UPDATED") return;
+    console.log("ChangesDetails.changes:", changes);
+    const prepChanges = Object.entries(changes).map(([key, change]) => {
+        if ("collection_of" in change) {
+            const added = change.collection_of == "User" ? usersToNames(change.added) : change.added;
+            const deleted = change.collection_of == "User" ? usersToNames(change.deleted) : change.deleted;
+            return {
+                propertyLabel: getPropertyLabel(historyItem.class_name, key),
+                isCollection: true,
+                added: added,
+                deleted: deleted,
+            };
+        } else if (key === "procurement_shop_id") {
```

Comment:
think a switch statement will make this a little easier to read

Reply:
see Tim's comment and my reply about some refactoring coming later